cli/azd/pkg/tools/azcli/mock_azcli.go

```diff
@@ -0,0 +1,211 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azcli
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerregistry/armcontainerregistry"
+)
+
+var _ AzCli = &Mock{}
+
+// Mock can be used to build a mock AzCli implementation based on provided functions
+type Mock struct {
```

Comment:
I feel like the mock generation generates just a bunch of extra source code that we don't really need.  Right now it is pretty easy to mock out any tool call including azcli using the mock exec command runner... You can setup different predicates as needed without having to run the actual infra tests each time.

As others mentioned I also find it hard to follow/debug with the nested/looped tests.

Reply:
> I feel like the mock generation generates just a bunch of extra source code that we don't really need. Right now it is pretty easy to mock out any tool call including azcli using the mock exec command runner... You can setup different predicates as needed without having to run the actual infra tests each time.

We no longer use the `az` CLI for these control plane operations anymore, due to your recent refactoring, so I think I can't use a mock command runner to do the validation I want.  Do you have a better idea for this?

Reply:
We chatted in person and Wallace pointed out I can mock the pipeline on AzCli and not have to do all this ceremony, which is great.

Reply:
@wbreza I rewrote this to do mocking at the HttpClient level and remove the new Mock based on functions type I had tried to add.